File: src/ts/tests/trajectory-articulation-key.test.ts
import { expect, test } from 'vitest';
import { Trajectory, Articulation } from '../model';

test('numeric articulation keys are normalized to decimals', () => {
  const art = new Articulation({ name: 'pluck', stroke: 'd' });
  const traj = new Trajectory({ articulations: { 0: art } });

  expect(traj.articulations['0.00']).toBeInstanceOf(Articulation);
  expect(traj.articulations['0']).toBeUndefined();
});

File: python/api/tests/trajectory_articulation_key_test.py
from python.api.classes import Trajectory, Articulation


def test_numeric_articulation_keys_are_normalized():
    art = Articulation({"name": "pluck", "stroke": "d"})
    traj = Trajectory({"articulations": {0: art}})

    assert isinstance(traj.articulations.get("0.00"), Articulation)
    assert "0" not in traj.articulations
